components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountConnectorImpl.java

```diff
@@ -560,4 +570,69 @@ private User getAssociatedUser(String associatedUserName) {
         associatedUser.setTenantDomain(tenantDomain);
         return associatedUser;
     }
+
+    private void populateRequiredUserClaims(int tenantId, List<UserAccountAssociationDTO> userAccountAssociations)
+            throws UserAccountAssociationException {
+
+        for (UserAccountAssociationDTO eachUserAccountAssociation : userAccountAssociations) {
+            Map<String, String> userClaims = getUserClaims(eachUserAccountAssociation, tenantId);
+            eachUserAccountAssociation.setFirstName(userClaims.get(LOCAL_CLAIM_URI_FIRST_NAME));
+            eachUserAccountAssociation.setLastName(userClaims.get(LOCAL_CLAIM_URI_LAST_NAME));
+            eachUserAccountAssociation.setEmail(userClaims.get(LOCAL_CLAIM_URI_EMAIL_ADDRESS));
+        }
+    }
+
+    private Map<String, String> getUserClaims(UserAccountAssociationDTO userAccountAssociationDTO, int tenantId)
+            throws UserAccountAssociationException {
+
+        Map<String, String> inboundAttributes = new HashMap<>();
+
+        RealmService realmService = IdentityAccountAssociationServiceComponent.getRealmService();
+        try {
+            UserRealm userRealm = realmService.getTenantUserRealm(tenantId);
+            if (!(userRealm instanceof org.wso2.carbon.user.core.UserRealm)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to get user claims with the user realm class: " + userRealm.getClass()
+                            .getName() + ". The user realm should be an instance of: "
+                            + org.wso2.carbon.user.core.UserRealm.class + ". Therefore returning empty attributes");
+                }
+                return inboundAttributes;
+            }
+
+            UserStoreManager userStore = ((org.wso2.carbon.user.core.UserRealm) userRealm).getUserStoreManager()
```

Comment:
what happens if userAccountAssociationDTO.getDomain()? Will it return PRIAMRY user store?

Reply:
+1. This should be handled as PRIMARY. Fixed in ed1ae3b5edbfd0ae2f2623f9ff52381e5aca9a99